### PR TITLE
split unit and integration tests in makefile

### PR DESCRIPTION
**Pull request by BenTheElder**
we're going to need to switch up the CI jobs, but this should be a better end state.

cc @aojea

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,7 +70,13 @@ install: build
 # ================================= Testing ====================================
 # unit tests (hermetic)
 unit:
```
Comment (BenTheElder) [inline Makefile:72]: ci runs this target
we want to rename the job and run the test target.

File: pkg/cluster/internal/providers/docker/network_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
+// +build !nointegration
```
Comment (BenTheElder) [inline pkg/cluster/internal/providers/docker/network_integration_test.go:1]: This allows us to run unit tests only without the annoying skip messages. we could drop the skip mechanism entirely, but it's not super onerous for now.

File: hack/make-rules/test.sh [hunk 3]
```diff
@@ -13,30 +13,49 @@
...
 go build -o "${REPO_ROOT}/bin/gotestsum" gotest.tools/gotestsum
 cd "${REPO_ROOT}"
 
+go_test_opts=(
+  "-coverprofile=${REPO_ROOT}/bin/${MODE}.cov"
+  '-covermode' 'count'
+  '-coverpkg' 'sigs.k8s.io/kind/...'
+)
+if [[ "${MODE}" = 'unit' ]]; then
+  go_test_opts+=('-run' '^TestIntegration')
```
Comment (aojea) [inline hack/make-rules/test.sh:38]: Are they flipped?

Comment (BenTheElder) [reply hack/make-rules/test.sh:38]: yes

**Pull request merged**